weekly_report/generate_report.py

```diff
@@ -229,6 +245,43 @@ def get_website_names(self) -> list[str]:
                 website_list.append(websites[x]['website_name'])
             return website_list
 
+    def get_tags_word_count(self):
+        """Returns the result of a sql query which returns tags and the
+          number of times it occurs in set of website ids.."""
+        with self.conn.cursor() as cur:
+            cur.execute("""SELECT t.tag_name, count(*) from tag AS t INNER JOIN game_tag_matching AS gt ON(t.tag_id=gt.tag_id) WHERE gt.game_id IN
+                        (SELECT game_id from game AS g WHERE g.release_date >= CURRENT_DATE - INTERVAL '7 days'
+                        AND g.release_date <= CURRENT_DATE AND g.website_id IN %s) GROUP BY t.tag_name ORDER BY count(*) DESC;
+                        """, (self.website_ids,))
+            tags_word_count = cur.fetchall()
+            return tags_word_count
+
+    def get_word_cloud(self):
```

Comment:
Good job on dividing functionality in this function; it is much easier to follow